### PR TITLE
Documentation: restructure dependencies for consistency

### PR DESCRIPTION
**Pull request by hmeine**
also, properly label links to installation instructions, so that
it is clear that they are for tensorflow / theano, not keras

### CODE REVIEW

**Comment by hmeine** [comment]
Did you close this on purpose?  Should I have elaborated?  Do you have any questions?

As is, I see the following problems (which I tried to solve with this PR):
1. The list of depencies is visually broken into parts, so that one would believe it to end before the first "When using …" line.
2. The "see installation instructions" are unqualified and on lines of their own, and they look exactly the same, so that they look like they point to instructions within the keras documentation.
3. HDF5 and h5py are marked as optional with a comment at the end, whereas cuDNN is only mentioned at the end of the line, because the "optional" part is at the beginning.

Those are all small things, but I do think that the PR is worth merging.

**Comment by hmeine** [comment]
This is to illustrate in particular the first issue:

bildschirmfoto 2016-12-05 um 13 43 22
